r_arm_controller:
  controller:
    package: bender_arm
    module: arm_controller
    type: ArmController
  joint_trajectory_action_node:
    min_velocity: 0.1
    constraints:
      goal_time: 0.25
    state_update_rate: 30
    joint_states_topic: 'r_arm/joint_states' # Local namespace
    operation_points:
      home:
        joint_names: [ 'r_shoulder_pitch_joint', 'r_shoulder_roll_joint' , 'r_elbow_pitch_joint']
        state: [0.0,0.0,0.0,0.0,0.0,0.0]
        params: [ [15, 5, 0],[15, 5, 0], [15, 5, 0]]
      pre_manipulation:
        joint_names: [ 'r_shoulder_pitch_joint', 'r_shoulder_roll_joint','r_elbow_pitch_joint']
        state: [0.0,0.0,0.0,1.2,0.0,0.0]
        params: [ [15, 10, 2], [15, 8, 2], [ 15, 10, 2]]
      manipulation:
        joint_names: [ 'r_shoulder_pitch_joint','r_shoulder_roll_joint', 'r_elbow_pitch_joint']
        state: [0.0,0.0,0.0,1.2,0.0,0.0]
        params: [ [15, 18, 5], [15, 10, 5], [ 16, 12, 5]]
    checkpoint_list:
      home:
        state: [0.0,0.0,0.0,0.0,0.0,0.0]
      pre_1:
        state: [0.0,0.0,0.0,1.5,0.0,0.0]
      pre_2:
        state: [0.0,0.0,0.0,1.5,0.0,0.0]
      carry:
        state: [0.0,0.0,0.0,1.5,0.0,0.0]
    joint_ops:
      shou_pitch:
        name: 'r_shoulder_pitch_joint'
        states: [-0.5,0.0,0.5,1.0]
        params: [[15, 8, 0],[15, 5, 0],[15, 8, 0],[17, 15, 0]]
      shou_roll:
        name: 'r_shoulder_roll_joint'
        states: [0.0,0.3,0.6,1.0]
        params: [[15, 5, 0],[15, 5, 0],[15, 5, 0],[15, 5, 0]]
      elb_pitch:
        name: 'r_elbow_pitch_joint'
        states: [0.0,0.5,1.0,1.5]
        params: [[15, 5, 0],[15, 5, 0],[15, 5, 0],[15, 5, 0]]


